R/tm_g_bivariate.R

```diff
@@ -665,47 +666,46 @@ srv_g_bivariate <- function(id,
       teal.code::eval_code(merged$anl_q_r(), substitute(expr = plot <- cl, env = list(cl = cl)))
     })
 
-    decorated_output_q <- srv_transform_teal_data("decorate", data = output_q, transformators = decorators)
-
-    decorated_output_q_facets <- reactive({
-      ANL <- merged$anl_q_r()[["ANL"]]
-      row_facet_name <- as.vector(merged$anl_input_r()$columns_source$row_facet)
-      col_facet_name <- as.vector(merged$anl_input_r()$columns_source$col_facet)
-
-      # Add labels to facets
-      nulled_row_facet_name <- varname_w_label(row_facet_name, ANL)
-      nulled_col_facet_name <- varname_w_label(col_facet_name, ANL)
-      facetting <- (isTRUE(input$facetting) && (!is.null(row_facet_name) || !is.null(col_facet_name)))
-      without_facet <- (is.null(nulled_row_facet_name) && is.null(nulled_col_facet_name)) || !facetting
-
-      print_call <- if (without_facet) {
-        quote(print(plot))
-      } else {
-        substitute(
-          expr = {
-            # Add facetting labels
-            # optional: grid.newpage() # nolint: commented_code.
-            # Prefixed with teal.modules.general as its usage will appear in "Show R code"
-            plot <- teal.modules.general::add_facet_labels(
-              plot,
-              xfacet_label = nulled_col_facet_name,
-              yfacet_label = nulled_row_facet_name
-            )
-            grid::grid.newpage()
-            grid::grid.draw(plot)
-          },
-          env = list(nulled_col_facet_name = nulled_col_facet_name, nulled_row_facet_name = nulled_row_facet_name)
-        )
-      }
-      decorated_output_q() %>%
-        teal.code::eval_code(print_call)
-    })
-
+    decorated_output_q_facets <- srv_decorate_teal_data(
+      "decorator",
+      data = output_q,
+      decorators = subset_decorators("plot", decorators),
+      expr = reactive({
```

Comment:
I'm torn including this long reactive in here. I had a feeling that `expr` from `srv_decorate_teal_data` is for short `prints` :)

Reply:
It was until it starts to be more complicated. There are 2 cases for this in {tmg}.

Open to have a better approach

Reply:
I think this is fine for now. Other approach would be to create this reactive and just pass it as an argument